src/config/settings.js

```diff
@@ -8,19 +8,51 @@ const fs = require('fs');
 const deepmerge = require('deepmerge');
 
 // Load the settings file
-function load() {
-
+/**
+ * @param{optional bool=false} inBrowser
+ * @returns {{kubernetes: {version: string}}}
+ */
+function load(inBrowser) {
+  if (typeof(inBrowser) == "undefined") {
+    inBrowser = false;
+  }
   // read the settings file into memory
   const rawdata = fs.readFileSync(paths.config() + '/settings.json');
   let settings = JSON.parse(rawdata);
-  let cfg = deepmerge(defaultSettings, settings);
-
-  // TODO: validate it
-
+  let cfg;
+  try {
+    verifyLocalSettings(settings);
+    cfg = deepmerge(defaultSettings, settings);
+  } catch(e) {
+    cfg = defaultSettings;
+    const header = "Error in saved settings.json file";
+    const msg = (/^Proposed kubernetes version.*not supported/.test(e.message)
+      ? `${e.message}. Using ${defaultSettings.kubernetes.version}`
+      : e.message);
```

Comment:
I was imaging the case where someone is using k8s 1.17.10 and somehow the settings got changed to 1.17.100. This isn't a valid version so it would cause 1.19.4 to be used instead. This would upgrade minikube. Once the problem is found the user can't correct it and bring k8s back up as 1.17.10 without wiping their content.

I know this is an edge case but we don't want to cause non-explicit destruction of data. This could lead to that by using the default.

Could we just do validation an not go with a default value? If validation fails not bring up k8s.

Reply:
Sounds good.

Reply:
This (reject bad version) was implemented in the second commit ("shutdown when settings contains an invalid k8s version") -- it shuts down because the settings file has been corrupted.

And speaking of the settings file, we should write it out with whitespace so it's more easily editable, in cases like these.  I have no problem making it easier to manually update dev-tool config files.

Reply:
And wait, hang on.... it's possible that minikube is invoked with an invalid version.

Reply:
whitespace in json file done